### PR TITLE
docs(theme): updated theme documentation

### PR DESCRIPTION
**Pull request by matthprost**
## Type

- Documentation

#### What is expected?

Updated theme documentation

### CODE REVIEW

File: src/__stories__/Documentation.stories.mdx [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import {
```
Comment (chambo-e) [inline src/__stories__/Documentation.stories.mdx:1]: @philibea I seem to remember that we can migrate these file to tsx since the last updates ?

Comment (matthprost) [reply src/__stories__/Documentation.stories.mdx:1]: @chambo-e I think we said we keep pure documentation in MDX as it is easier to read & write and only stories of components in tsx

Comment (chambo-e) [reply src/__stories__/Documentation.stories.mdx:1]: Afaik MDX prevents us from fully using react 17 automated runtime, maybe it's supported now

Comment (chambo-e) [reply src/__stories__/Documentation.stories.mdx:1]: Alright so apparently storybook is now able to have the new react runtime in mdx so I'll close this :)

**Review by chambo-e — APPROVED** [review]

**Review by QuiiBz — APPROVED** [review]

**Pull request merged**